pyiron_base/database/generic.py

```diff
@@ -463,6 +472,20 @@ def _check_chem_formula_length(self, par_dict):
             par_dict[key_limited] = "OVERFLOW_ERROR"
         return par_dict
 
+    def add(self, obj, settings, target_manager: Type[TableManager]):
+        if not self._viewer_mode:
+            try:
+                result = self.conn.execute(
+                    target_manager.add(obj, settings)
+                ).inserted_primary_key[-1]
+                if not self._keep_connection:
+                    self.conn.close()
+                return result
+            except Exception as except_msg:
+                raise ValueError("Error occurred: " + str(except_msg))
+        else:
+            raise PermissionError("Not avilable in viewer mode.")
+
     # Item functions
     def add_item_dict(self, par_dict):
```

Comment:
Doesn't the `add()` method replace `add_item_dict()`? I searched a bit through the repos in pyiron organizations, and found out that there might be couple of places, where we should replace `self.project.db.add_item_dict()` to `self.project.db.historical.add()`

Reply:
Since we already have `_check_chem_formula_length()` in the `HistoricalTable` class, I think that this is absolute.

Reply:
Yes, I will go through later this week to kill all the dead code and update the tests. In the current commits I only add and use the new functionality without removing the old ways.